Realm/RLMSyncPermissionResults.h

```diff
@@ -0,0 +1,112 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2017 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import <Foundation/Foundation.h>
+
+#import "RLMSyncUser.h"
+#import "RLMRealm.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// Properties which a sync permission results collection can be sorted by.
+typedef NS_ENUM(NSUInteger, RLMSyncPermissionResultsSortProperty) {
+    /// Sort by the path of the Realm the permission pertains to.
+    RLMSyncPermissionResultsSortPropertyPath,
+    /// Sort by the user ID of the user the permission applies to.
+    RLMSyncPermissionResultsSortPropertyUserID,
+};
+
+@class RLMSyncPermissionValue;
+
+/**
+ A collection object representing the results of a permissions query.
+
+ This collection will automatically update its contents at the start of each runloop
+ iteration, but the objects it vends are immutable and will themselves not change.
+
+ Permission results objects are thread-confined, and should not be shared across
+ threads.
+
+ @warning Permission results must only be fetched on threads that have an active
+          run loop. In most cases this will be the main thread.
+ */
+@interface RLMSyncPermissionResults : NSObject<NSFastEnumeration>
+
+/// The number of results contained within the object.
+@property (nonatomic, readonly) NSInteger count;
+
+/**
+ Return the first permission, or nil if the collection is empty.
+ */
+- (nullable RLMSyncPermissionValue *)firstObject NS_SWIFT_UNAVAILABLE("Use the `first` property.");
```

Comment:
Since these are marked as `NS_SWIFT_UNAVAILABLE`, maybe we should re-expose them in `RLMSupport.swift`? This applies to several of our existing APIs btw.